batch/batch/driver/instance.py

```diff
@@ -122,6 +126,19 @@ def adjust_free_cores_in_memory(self, delta_mcpu):
     def failed_request_count(self):
         return self._failed_request_count
 
+    async def check_is_active_and_healthy(self):
+        if self._state == 'active' and self.ip_address:
```

Comment:
Sorry, I misunderstood on first read.

I think `instances_by_last_updated` should be `live_instances_by_last_updated` (no need to check on invalid/deleted instances).  That can go in separately.  I agree with all the changes here.